sklearn/ensemble/tests/test_voting.py

```diff
@@ -681,3 +689,104 @@ def test_get_features_names_out_classifier_error():
     )
     with pytest.raises(ValueError, match=msg):
         voting.get_feature_names_out()
+
+
+# Metadata Routing Tests
+# ======================
+
+
+@pytest.mark.parametrize(
+    "Estimator, Child",
+    [(VotingClassifier, ConsumingClassifier), (VotingRegressor, ConsumingRegressor)],
+)
+def test_routing_passed_metadata_not_supported(Estimator, Child):
+    """Test that the right error message is raised when metadata is passed while
+    not supported when `enable_metadata_routing=False`."""
+
+    X = np.array([[0, 1], [2, 2], [4, 6]])
+    y = [1, 2, 3]
+
+    with pytest.raises(
+        ValueError, match="is only supported if enable_metadata_routing=True"
+    ):
+        Estimator(["clf", Child()]).fit(X, y, sample_weight=[1, 1, 1], metadata="a")
+
+
+@pytest.mark.usefixtures("enable_slep006")
+@pytest.mark.parametrize(
+    "Estimator, Child",
+    [(VotingClassifier, ConsumingClassifier), (VotingRegressor, ConsumingRegressor)],
+)
+def test_get_metadata_routing_without_fit(Estimator, Child):
+    # Test that metadata_routing() doesn't raise when called before fit.
```

Comment:
```suggestion
    """Test that get_metadata_routing() works regardless of the Child's
    consumption of any metadata."""
```

Reply:
The core of that issue was that this method would fail if the sub-estimator didn't support any metadata at all. So we test it here.

Reply:
I can see that in issue #28239 the routing failed on `fit`, when the sub-estimator cannot handle any sample weights.
Your solution from #28240 was to make sure `sample_weight` is getting to be part for `fit_params` before the routing is started and you added a test (`test_non_consuming_estimator_works`) that tests exactly that.
`VotingClassifier` is not covered by that test, but it surely needs a similar test. I still cannot see how `test_get_metadata_routing_without_fit` would test for the same case as `test_non_consuming_estimator_works`.

Reply:
the right PR: https://github.com/scikit-learn/scikit-learn/pull/28188/files